Code/Framework/AzFramework/AzFramework/Script/ScriptComponent.cpp

```diff
@@ -576,12 +581,18 @@ namespace AzFramework
     {
         DestroyEntityTable();
 
-        m_script = asset;
+        // Assign the updated asset on the next tick so we are not holding
+        // on to the old script
+        AZ::TickBus::QueueFunction(
+            [this, asset]()
+            {
+                m_script = asset;
 
-        if (LoadInContext())
-        {
-            CreateEntityTable();
-        }
+                if (LoadInContext())
+                {
+                    CreateEntityTable();
+                }
+            });
```

Comment:
I may have jumped the shipit gun too soon here.  But yeah, this is already being called on the main thread (its not the tick bus - its probably on a queue function being delivered as part of asset manager pump).

This _should_ work though as it would delay the execution of that block of code until the tick bus pumps next (which AFAIK is not the current tick, since its using the Asset Manager queue, not the tickbus queue IIRC...?).   The reason it should work is that the lambda itself captures 'asset'.  However, if you want the old asset to go out of scope you should probably set m_script to empty here, so that the old reference drops here?  Otherwise the old reference will only drop when the `m_script = asset` line hits.